### PR TITLE
Create ClosureBlog.md

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @joeduffy @lukehoban @mehzer @daveremy Feel free to edit in-place with any changes you think are appropriate!

File: reference/ClosureBlog.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+Many Cloud providers, like AWS, Azure and GCP, have all embraced the idea of providing [Function as a Service](https://en.wikipedia.org/wiki/Function_as_a_service) (FaaS) functionality to their platforms.  This functionality allows users of that platform to easily run logic for their cloud applications, without necessarily having to build or otherwise maintain the necessary underlying infrastructure for that logic themselves.  FaaS systems normally differ from more established [Platform as a Service](https://en.wikipedia.org/wiki/PaaS) (PaaS) systems in that they tend to be able to 'spin down' to using no cloud compute resources when not in use, while also being able to 'spin up' to respond to requests very quickly (ideally in milliseconds).  These systems are commonly called 'serverless' which, while being a complete misnomer (of course servers are involved!), does help convey the idea that in general you are free to squint and to not think about those servers *too much*.
```
Comment (mehzer) [inline reference/ClosureBlog.md]: Probably this whole para can just be dropped. I don't mind it as it's SEO juice, but means we have a lot of words up front. The other 3 paras are good philosophy though and I'd want to preserve most of that.

File: reference/ClosureBlog.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
... this.  We already wanted to supply an easy way for developers to simply spin up their cloud infrastructure using languages and tooling they were already familiar with.  However, when it came to FaaS offerings, things still felt a bit too onerous at first.  While you could certainly create an AWS Lambda with code like `new aws.lambda.Function("myfunc")`, you would still have to do all the work external to your Pulumi application to define your code, its dependencies, ways to pass necessary information into it, as well as potentially needing to have build tasks running to keep this all up to date.  That's a lot of extra, out-of-band, effort.  Doubly-so since a Pulumi application is *already* code!  Why have to jump out of that code just to create another bit of code that you need for running your Cloud application?  From a desire to do better here, the "closure serialized function" was borne.  However, "closure serialized function" is a pretty geeky mouthful, so I like just calling them "magic functions".  Before diving more deeply into them, let's take a look at how they look in action (note: some of this example depends on  API changes that will be present in an upcoming release):
```
Comment (mehzer) [inline reference/ClosureBlog.md]: Maybe trim this down?

At Pulumi, we thought there could be a better way to do this. From a desire to do better here, the "closure serialized function" was borne.  However, "closure serialized function" is a pretty geeky mouthful, so I like just calling them "magic functions".  Before diving more deeply into them, let's take a look at how they look in action (note: some of this example depends on  API changes that will be present in an upcoming release):

**Comment by CyrusNajmabadi** [comment]
Closing out as we've published this.

**Pull request closed without merging**